src/vmm/src/device_manager/mmio.rs

```diff
@@ -110,17 +108,32 @@ impl MMIODeviceManager {
     }
 
     /// Allocates resources for a new device to be added.
-    fn allocate_new_slot(&mut self) -> Result<MMIODeviceInfo> {
-        if self.irq > self.last_irq {
-            return Err(Error::IrqsExhausted);
+    fn allocate_new_slot(&mut self, irq_count: usize) -> Result<MMIODeviceInfo> {
+        let slot: MMIODeviceInfo;
+        match irq_count {
+            0 => {
+                slot = MMIODeviceInfo {
+                    addr: self.mmio_base,
+                    len: MMIO_LEN,
+                    irqs: vec![0],
+                };
+            }
+            1 => {
+                if self.irq > self.last_irq {
+                    return Err(Error::IrqsExhausted);
+                }
+                slot = MMIODeviceInfo {
+                    addr: self.mmio_base,
+                    len: MMIO_LEN,
+                    irqs: vec![self.irq],
+                };
+                self.irq += 1;
+            }
+            _ => {
+                return Err(Error::UnsupportedIrqCount);
+            }
```

Comment:
I'm fine with allowing any number of IRQs really:

```suggestion
        if self.irq + irq_count > self.last_irq + 1 {
            return Err(Error::IrqsExhausted);
        }
        let irqs = Vec::with_capacity(irq_count);
        for _ in 0..irq_count {
            irqs.push(self.irq);
            self.irq += 1;
        }
        slot = MMIODeviceInfo {
            addr: self.mmio_base,
            len: MMIO_LEN,
            irqs,
        };
```